tests/unit/cluster/cli.tcl

```diff
@@ -300,13 +300,10 @@ test {Migrate the last slot away from a node using redis-cli} {
         catch { $newnode_r get foo } e
         assert_equal "MOVED $slot $owner_host:$owner_port" $e
 
-        # Check that the empty node has turned itself into a replica of the new
-        # owner and that the new owner knows that.
-        wait_for_condition 1000 50 {
-            [string match "*slave*" [$owner_r CLUSTER REPLICAS $owner_id]]
-        } else {
-            fail "Empty node didn't turn itself into a replica."
-        }
+        # Check that the now empty primary node doesn't turned itself into
+        # a replica of any other nodes
+        after 5000
+        assert_equal [string match "*slave*" [$owner_r CLUSTER REPLICAS $owner_id]] 0
```

Comment:
There is an `assert_no_match` function to use for checks like this.

Btw, this code only checks that it's not a replica of the $owner node. Not "any over nodes" as said in the comment. We could check ROLE on the new node instead.